Задача №1  
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого 
кредитного скоринга (ks):

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy

Полученные значения должны быть равны.


In [49]:
import numpy as np

zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
cov = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
cov

9157.839999999997

In [50]:
cov0 = np.cov(zp, ks, ddof = 0)
cov0

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

Для смещенной ковариации:

In [51]:
zp0 = np.std(zp, ddof = 0)
ks0 = np.std(ks, ddof = 0)
r0 = 9157.84/(zp0 * ks0)
r0

0.8874900920739162

Для несмещенной ковариации:

In [52]:
cov1 = np.cov(zp, ks, ddof = 1)
cov1

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [53]:
zp1 = np.std(zp, ddof = 1)
ks1 = np.std(ks, ddof = 1)
r1 = 10175.37777778/(zp1 * ks1)
r1

0.88749009207411

 С использованием функции:

In [54]:
R = np.corrcoef(zp, ks)
R

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [55]:
import pandas as pd

arr = pd.DataFrame({'q': [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
 'w': [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]})

arr['q']. corr(arr['w'])

0.8874900920739163

Задача №2  
Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111.

Известно, что в генеральной совокупности IQ распределен нормально.

Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Находим среднее значение выборки:

In [56]:
import scipy.stats as st
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
sr = np.mean(x)
sr

118.1

Рассчитываем сигму по выборке:

In [57]:
mu = np.std(x, ddof=1)
mu

10.54566788359614

Находим t - критерий:

In [58]:
t = st.t.ppf(0.975, len(x)-1)
t

2.2621571627409915

In [59]:
[sr - t * (mu/(len(x)**0.5)), sr + t * (mu/(len(x)**0.5))]

[110.55608365158724, 125.64391634841274]

 Доверительный интервал для математического ожидания с надежностью 0.95 от 110.55608365158724 до 125.64391634841274

Задача №3  
Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [60]:
my = 25**0.5
i = 27
sr1 =174.2

Находим z - критерий:

In [61]:
z = st.norm.ppf(0.975)
z

1.959963984540054

In [62]:
[sr1-z*(my/(i**0.5)), sr1+z*(my/(i**0.5))]

[172.3140237765397, 183.99981992270025]

 Доверительный интервал для математического ожидания с надежностью 0.95 от 172.3140237765397 до 183.99981992270025